### PR TITLE
File integration

### PR DESCRIPTION
**Pull request by wtrocki**
Heavily inspired (aka copy paste) file integration docs.

### CODE REVIEW

File: modules/ROOT/pages/_partials/sync/js-client-files.adoc [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+//create a new client
+----
+
+= Uploading files from GraphQL 
+
+File upload capability adds new GraphQL scalar `Upload` that can be used for mutations that operate on binary data.
+`Upload` scalar maps html `FileList` HTML5 object in GraphQL schemas. 
+The first step required to work with binary uploads is to write mutation that will contain `Upload` scalar:
+
```
Comment (finp) [inline modules/ROOT/pages/_partials/sync/js-client-files.adoc]: The following example demonstrates how to upload a profile picture for the showcase app?

File: modules/ROOT/pages/_partials/sync/server-files.adoc [hunk 2]
```diff
@@ -0,0 +1,54 @@
+= File Upload Slient
```
Comment (finp) [inline modules/ROOT/pages/_partials/sync/server-files.adoc]: Client?

File: modules/ROOT/pages/_partials/sync/server-files.adoc [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+== Introduction
+
+The upload functionality using the GraphQL multipart form requests specification.
+The File Upload needs to be implemented on both server and client:
+
+1) On the client HTML FileList objects are mapped into a mutation and sent to the server in a multipart request.
+
+2) On the server: The multipart request is handled. The server processes it and provides an upload argument to a resolver. 
+In the resolver function, the upload promise resolves an object.
```
Comment (finp) [inline modules/ROOT/pages/_partials/sync/server-files.adoc]: would 1 and 2 be better in the intro?

Comment (finp) [reply modules/ROOT/pages/_partials/sync/server-files.adoc]: oh, it's already in the client side intro, doh!
so, so we need it in both places?

File: modules/ROOT/pages/_partials/sync/js-client-files.adoc [hunk 4]
```diff
@@ -0,0 +1,82 @@
+= File Upload Client
+
+{sync-client} provides support for uploading binary data along with the GraphQL queries.
+Implementation is relying on `apollo-upload-client` package build by Apollo community.
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/js-client-files.adoc]:
```suggestion
The file implementation uses the `apollo-upload-client` package built by the Apollo community.
```

File: modules/ROOT/pages/_partials/sync/js-client-files.adoc [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+== Introduction
+
+The upload functionality using the GraphQL multipart form requests specification.
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/js-client-files.adoc]:
```suggestion
The upload functionality uses the GraphQL multipart form requests specification.
```

**Comment by wtrocki** [comment]
@finp As a newbie could I get a credit of locations where links can be added?

File: modules/ROOT/pages/_partials/sync/server-files.adoc (hunk 3 above)
Comment (wtrocki) [reply modules/ROOT/pages/_partials/sync/server-files.adoc]: It is essential for the reader to know the premise, but not sure how to do that properly with the boundaries of server/client. Duplication is small so maybe we can close eye? Any suggestions welcome

Comment (finp) [reply modules/ROOT/pages/_partials/sync/server-files.adoc]: As part of sync docs review I'll try to merge, fine for now

File: modules/ROOT/pages/_partials/sync/js-client-files.adoc [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  fileUpload: true
+  ...
+};
+
+//create a new client
+----
+
+= Uploading files from GraphQL 
+
+File upload capability adds new GraphQL scalar `Upload` that can be used for mutations that operate on binary data.
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/js-client-files.adoc]:
```suggestion
File upload capability adds a new GraphQL scalar `Upload` that can be used for mutations that operate on binary data.
```

File: modules/ROOT/pages/_partials/sync/js-client-files.adoc [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+`Upload` scalar maps html `FileList` HTML5 object in GraphQL schemas.
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/js-client-files.adoc]:
```suggestion
The `Upload` scalar maps html `FileList` HTML5 object in GraphQL schemas. 
```

File: modules/ROOT/pages/_partials/sync/js-client-files.adoc [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+            ...UserFragment
+        }
+    }
+    `;
+----
+
+
+== Executing mutations
+
+`Upload` scalar will be mapped  to object returned from HTML file input.
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/js-client-files.adoc]:
```suggestion
The `Upload` scalar will be mapped  to object returned from HTML file input.
```

File: modules/ROOT/pages/_partials/sync/server-files.adoc [hunk 9]
```diff
@@ -0,0 +1,54 @@
+= File Upload Server
+
+{sync-server} provides support for uploading binary data along with the GraphQL queries.
+Implementation is relying on upstream `Apollo NodeJS` server capabilities
+
+== Introduction
+
+The upload functionality using the GraphQL multipart form requests specification.
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/server-files.adoc]:
```suggestion
The upload functionality uses the GraphQL multipart form requests specification.
```

File: modules/ROOT/pages/_partials/sync/server-files.adoc [hunk 10]
```diff
@@ -0,0 +1,54 @@
+= File Upload Server
+
+{sync-server} provides support for uploading binary data along with the GraphQL queries.
+Implementation is relying on upstream `Apollo NodeJS` server capabilities
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/server-files.adoc]: I'm not 100% sure but are you saying it relies on capabilities provided by the apollo-server package? If so my suggestion is this:

```suggestion
Implementation relies on upstream `Apollo Server` capabilities.
```

File: modules/ROOT/pages/_partials/sync/server-files.adoc [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+1) On the client HTML FileList objects are mapped into a mutation and sent to the server in a multipart request.
+
+2) On the server: The multipart request is handled. The server processes it and provides an upload argument to a resolver. 
+In the resolver function, the upload promise resolves an object.
+
+NOTE: File upload is based on link:https://github.com/jaydenseric/graphql-multipart-request-spec[graphql-multipart-request-spec]
+
+= Building Resolvers with File Uploads Support
+
+To enable file uploads please create a schema and using the `Upload` scalar.
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/server-files.adoc]:
```suggestion
To enable file uploads, create a schema and use the `Upload` scalar.
```

File: modules/ROOT/pages/_partials/sync/server-files.adoc [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+  type Query {
+    uploads: [File]
+  }
+  type Mutation {
+    singleUpload(file: Upload!): File!
+  }
+`;
+----
+
+Following schema enables file uploads. Resolvers will get injected `Upload` scalar as one of the arguments.
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/server-files.adoc]:
```suggestion
The following schema enables file uploads. The `Upload` scalar will be injected as one of the arguments in the resolvers.
```

File: modules/ROOT/pages/_partials/sync/server-files.adoc [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+`Upload` scalar will contain all metadata and also node.js stream that can be used to save file to specific location:
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/server-files.adoc]:
```suggestion
The `Upload` scalar contains all file metadata and a link:https://nodejs.org/api/stream.html#stream_readable_streams[Readable Stream] that can be used to save the file to a specific location.
```

File: modules/ROOT/pages/_partials/sync/js-client-files.adoc [hunk 14]
```diff
@@ -0,0 +1,83 @@
+= File Upload Client
+
+{sync-client} provides support for uploading binary data along with the GraphQL queries.
+The binary upload implementation uses the `apollo-upload-client` package built by the Apollo community.
+
+== Introduction
+
+The upload functionality uses the GraphQL multipart form requests specification.
+The File Upload needs to be implemented on both server and client:
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/js-client-files.adoc]:
```suggestion
The File upload needs to be implemented on both server and client:
```

File: modules/ROOT/pages/_partials/sync/js-client-files.adoc [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+const config = {
+  ...
+  fileUpload: true
+  ...
+};
+
+//create a new client
+----
+
+= Uploading files from GraphQL
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/js-client-files.adoc]:
```suggestion
= Uploading Files from GraphQL 
```

File: modules/ROOT/pages/_partials/sync/js-client-files.adoc [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+    }
+    `;
+----
+
+
+== Executing mutations
+
+The `Upload` scalar will be mapped  to object returned from HTML file input.
+
+Single file upload example for React client will look like follows:
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/js-client-files.adoc]:
```suggestion
The following example shows file upload in a React application.
```

File: modules/ROOT/pages/_partials/sync/server-files.adoc [hunk 17]
```diff
@@ -0,0 +1,54 @@
+= File Upload Server
+
+{sync-server} provides support for uploading binary data along with the GraphQL queries.
+Implementation relies on upstream `Apollo Server` capabilities.
+
+== Introduction
+
+The upload functionality uses the GraphQL multipart form requests specification.
+The File Upload needs to be implemented on both server and client:
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/server-files.adoc]:
```suggestion
File upload needs to be implemented on both server and client:
```

File: modules/ROOT/pages/_partials/sync/server-files.adoc [hunk 18]
```diff
@@ -0,0 +1,54 @@
+= File Upload Server
+
+{sync-server} provides support for uploading binary data along with the GraphQL queries.
+Implementation relies on upstream `Apollo Server` capabilities.
```
Comment (darahayes) [inline modules/ROOT/pages/_partials/sync/server-files.adoc]:
```suggestion
The implementation relies on upstream `Apollo Server` capabilities.
```

**Comment by wtrocki** [comment]
I guess I'm going to squash this now :)

**Comment by wtrocki** [comment]
@darahayes Thanks for a detailed review. General rules I got from the review:

- Keep Capitals for titles
- Use a/the for 
- Avoid common words
- Avoid too technical content and provide introduction

**Comment by finp** [comment]
@wtrocki not sure what you mean by 'Avoid common words', avoid overloading terms, eg client?

**Comment by wtrocki** [comment]
screen shot 2019-02-25 at 5 04 01 pm

**Comment by finp** [comment]
@wtrocki  ah, empty words, like empty calories ;)

**Pull request merged**